app/models/publishers/employer_enrollment_notification.rb

```diff
@@ -0,0 +1,110 @@
+module Publishers
+  class EmployerEnrollmentNotification
+
+    PublishError = Struct.new(:message, :headers)
+
+    attr_reader :employer
+
+    def initialize(employer)
+      @employer = employer
+    end
+
+    def employer_policies
+      policies = Policy.where(:employer_id => employer.id,
+                               :aasm_state.in => %w[submitted resubmitted effectuated],
+                              :carrier_id.in => enrollment_update_required_carrier,
+                              :enrollees =>
+                                   { "$elemMatch" => {"$and" => [
+                                       {"$or" => [
+                                           {"coverage_start" => { "$gt" => Date.today - 1.year }},
+                                           {"coverage_start" => { "$lt" => Date.today + 1.year }}
+                                       ]},
+                                       {"$or" => [
+                                           {"coverage_end" => nil},
+                                           {"coverage_end" => { "$gt" => Time.now }}]
+                                       }]}
+                                   })
```

Comment:
You can nix the AASM state check here.  AASM state on policies isn't.....strictly enforced.
 You're better off trusting your gut about the coverage start dates. Your second filter on line 28 that checks using the code rather than relying on just the database is 100% correct though.

However you need to make sure you are checking the dates of the SUBSCRIBER when doing the query - not just any enrollee.  The subscriber will be the enrollee that has "rel_code" = "self".

Reply:
Done!. Updated query to use subscriber rather relying on aasm state.